src/app/services/role-guard.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Router } from '@angular/router';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RoleGuardService {
+  constructor(private auth: AuthService, private router: Router) { }
+
+  // TODO: instead of redirecting to login, show a message to the user
```

Comment:
Not very important to me personally but it might confusing to just send the user the login page if they dont have sufficient permission.
We might want to instead send them to a forbidden page or similar.
(Perhaps status code page with get params).